src/ProphecyTrait.php

```diff
@@ -39,9 +39,15 @@ trait ProphecyTrait
      */
     protected function prophesize(?string $classOrInterface = null): ObjectProphecy
     {
-        if (\is_string($classOrInterface)) {
+        static $isPhpUnit9;
+        $isPhpUnit9 = $isPhpUnit9 ?? method_exists($this, 'recordDoubledType');
+
+        if ($isPhpUnit9 && \is_string($classOrInterface)) {
             \assert($this instanceof TestCase);
             $this->recordDoubledType($classOrInterface);
+        } else {
```

Comment:
This `else` clause is not always PHPUnit 10.1+ (there is a `&&` in the previous condition)

Reply:
Great catch! Fixed in 49b6444